### ISSUE TITLE
ucloud_disk: when using rssd, the rdma cluster is missing

### ISSUE THREAD

**Issue by fioncat** [ucloud/terraform-provider-ucloud#127]
When using the [ucloud_disk](https://registry.terraform.io/providers/ucloud/ucloud/latest/docs/resources/disk) resource to create the rssd disk, terraform is missing the rdma information of the rssd disk, and the backend will randomly assign a rdma cluster to it.

In this way, the rdma cluster of uhost may be inconsistent with that of udisk, and an error will be reported when binding the disk:

```text
Error: error on creating disk attachment, api:
[server.RetCodeError] 17218 uhost error, Error: Command failed: add_udisk.py failed
```

Now we have 2 solutions:

- Add a field to `ucloud_disk` so that user can specify `rdma_cluster_id`.
- Forbid users to create rssd disks in this way, forcing them to use [data_disk](https://registry.terraform.io/providers/ucloud/ucloud/latest/docs/resources/instance#data_disks) in [ucloud_instance](https://registry.terraform.io/providers/ucloud/ucloud/latest/docs/resources/instance).

**Comment by leason00** [comment]
I prefer the first solution, ,and rdma_cluster_id can be optional. Because you can't restrict users from creating cloud disks alone.

**Comment by fioncat** [comment]
> I prefer the first solution, ,and rdma_cluster_id can be optional. Because you can't restrict users from creating cloud disks alone.

Sorry for the delay, we will add a field, please wait for our fix, thx.

**Comment by leason00** [comment]
> * rdma_cluster_id

It is best to obtain the newly added fields automatically, such as from the returned result of host, or directly specify the host ID. thx

**Comment by fioncat** [comment]
> > * rdma_cluster_id
> 
> It is best to obtain the newly added fields automatically, such as from the returned result of host, or directly specify the host ID. thx

Yes, it will be returned by ucloud_instance, for example:

```terraform
# Create a web server
resource "ucloud_instance" "web" {
  ...
}

# Create cloud disk
resource "ucloud_disk" "default" {
  # The rdma cluster id will be returned by resource ucloud_instance
  rdma_cluster_id = ucloud_instance.web.rdma_cluster_id
}
```

**Comment by fioncat** [comment]
Fixed, please update provider to [v1.32.0](https://github.com/ucloud/terraform-provider-ucloud/releases/tag/v1.32.0)

**Comment by leason00** [comment]
> Fixed, please update provider to [v1.32.0](https://github.com/ucloud/terraform-provider-ucloud/releases/tag/v1.32.0)

https://registry.terraform.io/providers/ucloud/ucloud/latest/docs/resources/instance. Is it convenient to update the document?

**Comment by fioncat** [comment]
> > Fixed, please update provider to [v1.32.0](https://github.com/ucloud/terraform-provider-ucloud/releases/tag/v1.32.0)
> 
> https://registry.terraform.io/providers/ucloud/ucloud/latest/docs/resources/instance. Is it convenient to update the document?

The doc went wrong, I will release v1.32.1 to fix the doc, sry.

**Comment by fioncat** [comment]
> > Fixed, please update provider to [v1.32.0](https://github.com/ucloud/terraform-provider-ucloud/releases/tag/v1.32.0)
> 
> https://registry.terraform.io/providers/ucloud/ucloud/latest/docs/resources/instance. Is it convenient to update the document?

Done:

- doc: [ucloud_instance#rdma_cluster_id](https://registry.terraform.io/providers/ucloud/ucloud/latest/docs/resources/instance#rdma_cluster_id), [disk#rdma_cluster_id](https://registry.terraform.io/providers/ucloud/ucloud/latest/docs/resources/disk#rdma_cluster_id)
- example: [rssd example](https://github.com/ucloud/terraform-provider-ucloud/tree/master/examples/rssd)

If you have futher problem, please contact our SPT or directly draft a new issue in this repo, thanks.
